### PR TITLE
Collect container last state

### PR DESCRIPTION
**Pull request by daweim0**
This PR adds the ContainerLastStatus field to KubePodInventory. It writes the fields lastState, reason, startedAt, and finishedAt from the `kubectl describe pod` command into ContainerLastStatus (as a Json object). If a pod has no last state then `{}` is written into the ContainerLastStatus field.

(tagging @ganga1980)

### CODE REVIEW

File: source/code/plugin/in_kube_podinventory.rb [hunk 1]
```diff
@@ -317,6 +317,19 @@ def parse_and_emit_records(podInventory, serviceList)
                   record["ContainerStatusReason"] = containerStatus[containerStatus.keys[0]]["reason"]
                 end
               end
+
+              # Record the last state of the container. This may have information on why a container was killed.
+              if container["lastState"].keys.length > 0
```
Comment (vishiy) [inline source/code/plugin/in_kube_podinventory.rb]: can u check if "lastState" always exist ? What do you see for a newly created deployment for example where there is no 'last' state?

File: source/code/plugin/in_kube_podinventory.rb [hunk 2]
```diff
@@ -317,6 +317,19 @@ def parse_and_emit_records(podInventory, serviceList)
...
+                lastStateName = container["lastState"].keys[0]
+                record["ContainerLastStatus"] = Hash.new
```
Comment (vishiy) [inline source/code/plugin/in_kube_podinventory.rb]: Again, you are assuming all these properties exist all the time. Is that true ? Can you please add check for existence ?

File: source/code/plugin/in_kube_podinventory.rb [hunk 3]
```diff
@@ -317,6 +317,19 @@ def parse_and_emit_records(podInventory, serviceList)
                   record["ContainerStatusReason"] = containerStatus[containerStatus.keys[0]]["reason"]
                 end
               end
+
+              # Record the last state of the container. This may have information on why a container was killed.
+              if container["lastState"].keys.length > 0
+                lastStateName = container["lastState"].keys[0]
```
Comment (vishiy) [inline source/code/plugin/in_kube_podinventory.rb]: keys[0] exists all the time ?

File: source/code/plugin/in_kube_podinventory.rb (hunk 1 above)
Comment (vishiy) [inline source/code/plugin/in_kube_podinventory.rb]: For safer side, can you put this entire block under rescue so if this fails it causes no impact ?

Comment (daweim0) [reply source/code/plugin/in_kube_podinventory.rb]: Yes.

**Pull request merged**